From 9e137b1b9481cd97a9b4b445aa573e0080086e2b Mon Sep 17 00:00:00 2001
From: Eliad Peller <eliad@wizery.com>
Date: Mon, 2 Jul 2012 15:08:25 +0300
Subject: [PATCH 16/40] mac80211: always set in_reconfig=false on wakeup

If the interfaces were removed just before a restart
work was started, open_count will be 0, and most of
the reconfig work will be skipped, including the
resetting of local->in_reconfig to false.

Leaving local->inconfig = true will result in
dropping any incoming packet.

Fix it by always setting local->in_reconfig = false
(even if there are no active interfaces).

Signed-off-by: Eliad Peller <eliad@wizery.com>
Signed-off-by: Johannes Berg <johannes.berg@intel.com>
---
 drivers/net/wireless/compat-wireless_R5.SP2.03/net/mac80211/util.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/net/wireless/compat-wireless_R5.SP2.03/net/mac80211/util.c b/drivers/net/wireless/compat-wireless_R5.SP2.03/net/mac80211/util.c
index 370a7fe..1c86742 100644
--- a/drivers/net/wireless/compat-wireless_R5.SP2.03/net/mac80211/util.c
+++ b/drivers/net/wireless/compat-wireless_R5.SP2.03/net/mac80211/util.c
@@ -1320,10 +1320,10 @@ int ieee80211_reconfig(struct ieee80211_local *local)
 		if (ieee80211_sdata_running(sdata))
 			ieee80211_enable_keys(sdata);
 
+ wake_up:
 	local->in_reconfig = false;
 	barrier();
 
- wake_up:
 	/*
 	 * Clear the WLAN_STA_BLOCK_BA flag so new aggregation
 	 * sessions can be established after a resume.
-- 
1.7.0.4

